### PR TITLE
Fixes login modal spacing

### LINKED ISSUE #6933: Login modal styling inconsistencies
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### The Problem
As you can already see, something is not right about this login modal!
It's styling! Some elements are spaced too much and some elements are squished together.

modal2

Your task is to fix it by making it styles consistent, something like this.
You can also introduce you own ideas if you think that you can do better ;)

modal

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file plots2/app/assets/stylesheets/login_modal.scss, use margin to space out elements.

CSS that needs to be implemented:

```css
div.form-group {
    margin-top: 1rem;
}

div#socialIcons {
    margin-bottom: -2rem;
}

p.form-grey {
    margin-top: 1rem;
}
```

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by yeo-yeo**
Fixes #6933 

Sorry it was slow, I had some issues with my setup
FYI @publiclab/reviewers
my twitter is @yeo_yeo_ !

Thanks for providing this opportunity 😊 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation (N/A?)
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by yeo-yeo** [comment]
@publiclabs/reviewers

**Comment by yeo-yeo** [comment]
FYI @VladimirMikulic , not sure why the @publiclabs/reviewers handle doesn't link

**Comment by fluctix** [comment]
Can you add a screenshot of the changes?

**Comment by yeo-yeo** [comment]
Thanks for the feedback.  I've put the #socialicons modifiers together, and also realised the #visibility lines could be put together too.

On returning to the request, I realised that the eye icon was not vertically centred and the space below the social media buttons was still large, so I made some further changes to address this - screenshots below.

Screenshot of original view:
original

Screenshot after first changes (i.e. original pull request):
1st suggested fix

Screenshot after second changes (i.e. current state):
2nd suggested fix

Let me know what you think! Thanks for you patience

**Comment by yeo-yeo** [comment]
Oops, sorry! Fixed.

RE the line-height, the original code seems to be conflicting since a#visibility had line-height 0, but #visibility had line-height 2.2em.  I chose 1.5em to be consistent with the password input it sits next to.  What do you think's best? Perhaps I've missed the reason for the original setting.

Will push on your feedback, with a new screenshot.  Thanks again 😊

**Comment by VladimirMikulic** [comment]
Originally line-height for #visibility was set to 2.2em, then I've set it to 0.
It was causing issues with view password element being too big.

big

If they decide to change font-size in the future we'll have the same problem again, just revert line-height back to 0.

Thank you for cooperation and patience.

**Comment by yeo-yeo** [comment]
Ok thanks, I've reverted to the original code settings for line-height (see screenshot from current live code:
Screenshot 2019-12-08 at 18 05 17
)

Screenshot of local view after these changes:
Screenshot 2019-12-08 at 18 08 47

As discussed, although the eye icon is not vertically centred, it seems that this is nonetheless displaying correctly in production so no need to change

Thanks!

File: app/assets/stylesheets/login_modal.scss [hunk 1]
```diff
@@ -8,17 +8,21 @@
...
-div.form-group {
-  margin-top: 1rem;
+  p.form-grey {
+    margin-top: 1rem;
+  }
 }
 
-p.form-grey {
-  margin-top: 1rem;
+#visibility {
```
Comment (VladimirMikulic) [inline app/assets/stylesheets/login_modal.scss]: I've just realized now that you can also put visibility in .login-form(like you did originally) and replace 2.2em with 0:

```css
.login-form {
    #visibility{
        color: #888;
        pointer-events: auto;
        line-height: 0;
        text-decoration: none;
    }
}
```

**Comment by yeo-yeo** [comment]
Ok I've re-pushed!

Screenshot (FWIW):
reorganise visibility selector

We'll get there eventually 😄

**Comment by VladimirMikulic** [comment]
@yeo-yeo Awesome you did it. Congratulations, you should be proud of yourself because you did something that matters 🎉🎉🎉
I know, it's been tricky, but we all make mistakes that's how we learn :)

@SidharthBansal & @jywarren, PR is ready to be merged.

**Comment by yeo-yeo** [comment]
@VladimirMikulic thank you so much for your patience!  It was my very first open source PR.  I thought I'd chosen something super easy but it goes to show things always take longer than you think 😉
It was super useful for learning about how process works, and what to look for when updating existing code.  Thanks again! 🎆

**Comment by VladimirMikulic** [comment]
Wow, I didn't know that this is your first PR 👏👏👏
That's how things work, it usually takes a few iterations until it's finally done.
It's not just you :)

**Comment by SidharthBansal** [comment]
@VladimirMikulic I will be really grateful if you can help other GCI students to have detailed reviews like you do.
@yeo-yeo Great work. 🚀 
I am loving it ❤️ 
Can you please start doing some help wanted issues?
I mean issues without `fto-candidate` and without `first-timers-only` label.
If you are a GCI student, then @VladimirMikulic can guide you to PL community, welcoming post, initial gci student duties which involves installation+2fto+supportive tasks+weekly pr reviews+ other issues(hard/medium/hall of fame).
Thanks both of you.

**Comment by VladimirMikulic** [comment]
@SidharthBansal I'll be glad to help other students with review process.
It is the quality that matters.

**Pull request merged**